source/type/tests/test.cpp

```diff
@@ -141,6 +144,47 @@ BOOST_AUTO_TEST_CASE(projection) {
     BOOST_CHECK_CLOSE(d, pp.distance_to(p), 1e-3);
 }
 
+std::shared_ptr<StopArea> create_stop_area (std::string const &name) {
+    auto sa = std::make_shared<StopArea>();
+    sa->name = name;
+    return sa;
+}
+
+bool operator==(const navitia::type::StopArea& sa, const std::string& sa_name) {
+    return sa.name == sa_name;
+}
+
+bool operator!=(const navitia::type::StopArea& sa, const std::string& sa_name) {
+    return !(sa.name == sa_name);
+}
+
+BOOST_AUTO_TEST_CASE(human_sort_on_stop_areas_should_not_throw) {
+    std::vector<std::shared_ptr<StopArea>> stop_areas {
```

Comment:
To wrap up with the `unique_ptr` story :D

Turns out, when initialising a `std::vector` of with an `initializer_list`, a **copy** of the initializer list is performed at the vector's constructor level : [libstdc++/stl_vector.h](https://github.com/gcc-mirror/gcc/blob/gcc-9_2_0-release/libstdc++-v3/include/bits/stl_vector.h#L611-L628).... And copying a `std::unique_ptr` is forbidden :no_good_man: :no_entry_sign: :no_good_man: 

The way to do it with `unique_ptr` would have been to use a `std::array` like in - https://godbolt.org/z/usFACF

Reply:
Alright! Should I do it that way with `unique_ptr` using `std::array` or leave it as it is right now ?

Reply:
You can leave it as it it. It's just for our education ;)

Reply:
Thanks for the heads up :)